python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -122,3 +124,94 @@ def preprocess_datasets(self) -> None:
         # This is only applicable for xgboost-ray<0.1.16
         if Version(xgboost_ray.__version__) < Version("0.1.16"):
             self._repartition_datasets_to_match_num_actors()
+
+
+class SimpleXGBoostTrainer(DataParallelTrainer):
+    def __init__(
+        self,
+        train_loop_per_worker: Union[Callable[[], None], Callable[[Dict], None]],
+        *,
+        train_loop_config: Optional[Dict] = None,
+        xgboost_config: Optional[XGBoostConfig] = None,
+        scaling_config: Optional[ray.train.ScalingConfig] = None,
+        run_config: Optional[ray.train.RunConfig] = None,
+        datasets: Optional[Dict[str, Any]] = None,
+        dataset_config: Optional[ray.train.DataConfig] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+    ):
+        if not xgboost_config:
+            xgboost_config = XGBoostConfig()
+
+        dataset_config = dataset_config or ray.train.DataConfig()
+        dataset_config._convert_to_data_iterator = False
+
+        super(SimpleXGBoostTrainer, self).__init__(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config=train_loop_config,
+            backend_config=xgboost_config,
+            scaling_config=scaling_config,
+            dataset_config=dataset_config,
+            run_config=run_config,
+            datasets=datasets,
+            resume_from_checkpoint=resume_from_checkpoint,
+            metadata=metadata,
+        )
+
+
+if __name__ == "__main__":
+    from contextlib import contextmanager
+
+    @contextmanager
+    def ray_train_xgboost_setup():
+        # env_vars = {
+        #     "DMLC_NUM_WORKER",
+        #     "DMLC_TASK_ID",
+        #     "DMLC_TRACKER_URI",
+        #     "DMLC_TRACKER_PORT",
+        # }
+
+        # rabit_args = {k: os.environ[k] for k in env_vars}
+
+        from xgboost.collective import CommunicatorContext
+
+        with CommunicatorContext():
+            yield
+
+    def train_fn(config):
+        from xgboost.collective import CommunicatorContext
+
+        with CommunicatorContext():
```

Comment:
Is this needed?

Reply:
Yup, this is the thing that actually connects the worker to the collective group (kind of like init_process_group).

I was trying to do it for the user [here](https://github.com/ray-project/ray/pull/42767/files/d8ecfa8ffefa0c1da25c4b6540f993e12a18d1c7#diff-e40514dc5fac1d96905235c5090fbeeb747889a95a3dd86228f8737348911236R72-R78), but that didn't end up working, since the context needs to be directly wrapping the user code it seems.

Reply:
Oh I see. @woshiyyya brought up a similar thing for Torch where we might want to set the torch device, which also needs to modify the user code since it has to be run in the same thread.

Maybe we can have some sort of decorator abstraction that surrounds the user's train function?

Reply:
Yes. I was trying to set torch cuda device by default, but it only works when we call it inside the training function. I am thinking we can have a function decorator, so that we can inject some environment setup in an elegant way.

e.g.
```
@ray.train(framework="torch")
def train_func():
    ...

```

Reply:
Seems that it's more crucial for the new XGBoostTrainer API. We could consider having this decorator so users don't have to think about it.

Reply:
Interesting, we may be able to get rid of the `Trainer`s and mirror the Ray Core API more. 😆